### PR TITLE
Refactor config files

### LINKED ISSUE #260: Refactor config folder
Our current structure looks like this:

```
config
├── accessibility.php
├── block-editor-settings.php
├── child-theme-settings-genesis.php
├── child-theme-settings.php
├── color-palette.php
├── custom-logo.php
├── html5.php
├── import
├── menus.php
├── onboarding.php
├── responsive-menus.php
└── simple-social-icons-settings.php
```

Before the config folder continues to grow, we should consider moving toward a more simple structure like this:

```
config
├── defaults.php // was child-theme-settings.php
├── appearance.php // fonts and colors-> block-editor-settings.php, color-palette.php
├── theme-supports.php // custom-logo.php, html5.php, genesis-accessibility.php, menus.php 
├── import
├── onboarding.php
├── responsive-menus.php
└── simple-social-icons.php // drop '-settings' from the name
```

The main changes are:

- Move all theme-supports into a single file. (This could potentially then be autoloaded by Genesis in the future, removing all `add_theme_support` lines from child themes that want to use the feature.)
- Remove `block-editor-settings.php` in favor of `appearance.php` (or `design.php` or `styles.php`?), where all fonts and colors can be declared.
- Remove '-settings' from filenames, given that all of them contain settings of some kind or other.

Edit:

Noting that `child-theme-settings.php` will have to continue to use that name, as it's hard-coded and auto-loaded by Genesis: https://github.com/studiopress/genesis/blob/8899887473437ba5a0a017b0214da9e17a009677/lib/admin/onboarding/theme-activation.php#L53

### PR DESCRIPTION
**Pull request by nickcernis**
Fixes #260.

- I had to keep `child-theme-settings.php` instead of renaming it to `defaults.php` as intended, because it's hard-coded and auto-loaded by Genesis: https://github.com/studiopress/genesis/blob/8899887473437ba5a0a017b0214da9e17a009677/lib/admin/onboarding/theme-activation.php#L53
- I removed `genesis_sample_get_default_color()` in favor of fetching the default colors from the config directly. (We don't have getter functions for other configuration data, so it seems unnecessary to have one for the colors.)

### CODE REVIEW

**Comment by nickcernis** [comment]
Noting two potential improvements for the future (that each require changes in Genesis):

- Move `responsive-menus.php` into `theme-supports.php` (would require Genesis to allow `genesis_register_responsive_menus` support to be declared via a theme support instead of the current function call, which would make more sense to me anyway).
- Remove `genesis_sample_theme_support` from `functions.php` and have Genesis autoload the theme-supports.php file

File: config/appearance.php [hunk 1]
```diff
@@ -28,7 +38,10 @@
 		array(
 			'name'  => __( 'Accent color', 'genesis-sample' ),
 			'slug'  => 'theme-secondary',
-			'color' => get_theme_mod( 'genesis_sample_accent_color', genesis_sample_get_default_color( 'accent' ) ),
+			'color' => get_theme_mod(
+				'genesis_sample_accent_color',
+				$genesis_sample_default_colors['accent']
```
Comment (dreamwhisper) [inline config/appearance.php]: Should this be defined earlier like the link color for consistency?
```php
$genesis_sample_accent_color = get_theme_mod(
	'genesis_sample_accent_color',
	$genesis_sample_default_colors['accent']
);
```

File: config/appearance.php [hunk 2]
```diff
@@ -18,6 +27,7 @@
 	'default-button-bg'            => $genesis_sample_link_color,
```
Comment (dreamwhisper) [inline config/appearance.php]: Doesn't have to be for this PR, but for each `default-` here, they aren't always "default" once the colors are changed. Perhaps we should look at renaming these?

**Comment by nickcernis** [comment]
@dreamwhisper Thanks for your review. I like both changes you suggested and have updated this PR.

**Pull request merged**